.github/workflows/python-package.yml

```diff
@@ -93,7 +93,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest, macos-latest]
-        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
+        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11', '3.12']
```

Comment:
I think this will require a bump of Pybind11: https://github.com/pybind/pybind11/releases/tag/v2.11.0. From what I see in the test it seems to work with the version we currently use, but it would be preferable to use a version that is known to work and officially works with 3.12.

Reply:
Hi ! 
pybind11 seems to be installed via a submodule who point to the repo of pybind11, so I think it's always taking the last commit of the master branch. 
See here : https://github.com/AcademySoftwareFoundation/OpenTimelineIO/blob/main/.gitmodules

Reply:
That's not how submodules work. Submodules are pinned to a specific commit.

Reply:
Yeah I didn't remember ! 
I have updated the submodule for pybind11 to the latest tag (v2.11.1)